project/frontend/src/components/InvestmentChart/InvestmentChart.js

```diff
@@ -0,0 +1,100 @@
+import React from "react";
+import ChartCard from "../../components/ChartCard/ChartCard";
+
+const InvestmentChart = (props) => {
+    const ONE_WEEK_MILLISECONDS = 7 * 24 * 60 * 60 * 1000;
+    const ONE_DAY_MILLISECONDS = 24 * 60 * 60 * 1000;
+
+    const getEarliestDate = (investments) => {
+        var currentEarliest = Infinity;
```

Comment:
following on from the previous comment. I was actually trying to make 2 points.
- investments is a collection. Can the collection be empty, if it can be you will return Infinity - ONE_WEEK_MILLI (will the client code of this function handle that? ). If a collection can't be empty, you can initialize currentEarliest = investments[0] dateInvestedMilliSeconds
- The function overall bakes in an offset to currentEarliest (ie ONE_WEEK_MILLISECONDS). This seems to overload its purpose. Suggestion is to subtract the offset in the code that calls this function OR rename the function to add some context/meaning to what ONE_WEEK_MILLISECONDS is achieving.

Reply:
It was probably a poor choice of variable naming in combination with the function name. The function gets the earliest date in the database (which is always 1 week before the buy in date as we fetch context data), and currentEarliest is a tracker for the earliest buy in date as we loop through. I'll change both names to be more clear, and add a docstring to explain the offset.

Reply:
As for the investments length, the function is only called in an if statement that checks for a non-empty array. I'll add a comment to the function that makes that explici.